### ISSUE TITLE
Add test resource "restore-all" into syndesis-rest

### ISSUE THREAD

**Issue by jludvice** [syndesisio/syndesis-e2e-tests#41]
There is rest resource for e2e tests. This endpoint is enabled in test environment,
but disabled in production.

Source:
https://github.com/syndesisio/syndesis-rest/blob/master/rest/src/main/java/io/syndesis/rest/v1/handler/tests/TestSupportHandler.java#L44

It has resource for restoring database. Doing so will delete integrations and connections,
but it leaks all running openshift pods, deployment configs...

There is also [OpenshiftService](https://github.com/syndesisio/syndesis-rest/blob/master/openshift/src/main/java/io/syndesis/openshift/OpenShiftService.java) which might be of some use and I'd expect some injectable bean for listing existing integrations. Not sure where it is thought.

**Comment by jludvice** [comment]
Here is my curl to test restore-db resource:
```sh
#!/usr/bin/env bash


HOST="https://syndesis-qe.b6ff.rh-idev.openshiftapps.com"
TOKEN="take access token from browser session storage"

#fetch data
curl  -X GET -v \
	-H "Content-Type: application/json" -H "Authorization: Bearer $TOKEN" "${HOST}/api/v1/test-support/snapshot-db"
```

**Comment by dsimansk** [comment]
As discussed with @jknetl this morning. There's a problem in the impl https://github.com/syndesisio/syndesis-rest/pull/491 with usage of correct token to delete the deployments. That may result in 500 HTTP error response during reset process.

**Comment by dsimansk** [comment]
@jknetl after applying fix 483a9285ba2c0d9aa210f9ddb099510d69b3e833 the issue on CI job was fixed. I'll mark this task as Done, thanks.

Done via syndesisio/syndesis-rest#491
